stustapay/core/service/order.py

```diff
@@ -38,12 +57,21 @@ async def create_order(
                 "where product.id = $1;",
                 item_id,
             )
-
             if cost is None:
-                raise Exception("product not found")
-
+                raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"Product {item_id} not found")
             price, tax_rate, tax_name = cost
 
+            # check age restriction
+            restricted = await conn.fetchval(
+                "select * from product_restriction r join product p on r.id = p.id where r.restriction = $1",
+                customer["restriction"],
+            )
+            if restricted is not None:
+                raise HTTPException(
+                    status_code=status.HTTP_403_FORBIDDEN, detail=f"Product {item_id} is age restricted"
```

Comment:
if this http code is useful? i'm always confused when (especially intended) programmatic errors cause http errors.